### DISCUSSION TITLE
Include AppImageLauncher by default

### DISCUSSION THREAD

**Discussion by VewDev** [linuxmint/discussions#681] [Ideas]
I think AppImageLauncher (or a modified version with a more friendly UI) should be included in Mint by default, especially on Cinnamon.

I know a lot of people may dislike AppImages for a variety of reasons, including integration and package duplication.

However, I feel like AppImages align pretty well with Mint's "ease of use" philosophy, since they allow users to quickly launch applications from an icon on their desktop. And to be honest, a lot of people don't know the difference between a shortcut to an app and an AppImage itself.

If you wanted to avoid using AppImages in your Mint installation, you could just not download any AppImages, and the AppImageLauncher app wouldn't bother you (you can also uninstall AppImageLauncher if you really dislike it).

I think this would give Linux newcomers choosing Mint as their first distro a more welcoming and smooth experience.

This is related but not the exact same as #188 

AppImageLauncher source link: https://github.com/TheAssassin/AppImageLauncher

**Comment by Secret-chest** [comment]
Just add them to the menu, but not to the desktop.

**Reply by VewDev** [reply]
what do you mean by this?

**Reply by contrarybaton60** [reply]
Secret-chest meant to manually add a .desktop file with the executable linking to the AppImage

**Comment by Secret-chest** [comment]
I actually like it.

**Comment by SuperninjaXII** [comment]
Yeah this will be good

**Comment by SpeeQz1** [comment]
[AppImageLauncher](https://github.com/TheAssassin/AppImageLauncher) isn't being maintained much and it's already showing problems with AppImages from apps like FreeCAD due to `zstd` compression FreeCAD/FreeCAD-Bundle#331. I would recommend [Gear Lever](https://github.com/mijorus/gearlever) be used instead.

**Reply by Secret-chest** [reply]
Gear Lever uses libadwaita, so it doesn't integrate nicely visually. We could make our own launcher though since it isn't that complicated.

**Reply by VewDev** [reply]
Making a custom launcher is a lot more work than simply including a reskinned AppImageLauncher (or Gear Lever or whatever) in the default installation. Also see: https://xkcd.com/927/

**Reply by contrarybaton60** [reply]
Still, AppImageLauncher uses Qt so it still wouldn't make sense.

**Reply by Secret-chest** [reply]
We could try to separate the AIL backend and make a new GTK interface.

**Reply by VewDev** [reply]
After doing a little research, I agree that AppImageLauncher is arguably getting less maintainance than Gear Lever. The compressions issues seem to be solved in the development version though ( https://github.com/TheAssassin/AppImageLauncher/issues/674) so one would expect them to be ready in the production version soon.

Anyway, I also do agree that the UI for AppImageLauncher NEEDS to be reworked in order to be integrated into LM, in fact, I already said "a modified with a more friendly UI" would be a good idea in the original post.

That being said, I personally don't have anything against Gear Lever, it's just that I haven't *personally* used it, so wouldn't know how well it works. If Gear Lever is a better option for LM integration in general, then definitely go with it.

That also applies to any other AppImage "launchers" I might not know about.

**Comment by Secret-chest** [comment]
I think it would be better to move all AppImages to a special directory (/usr/share/appimages, ~/.appimages)

**Comment by AtomicRobotMan0101** [comment]
Came here today to make this EXACT recommendation.

This week Ive migrated one noob off Windows and another off MacOS and both were stuck after downloading an appImage.

By having a **trivial launcher which asks the user WHAT they'd like to do**, for this very specific thing, seems like a bloody good idea.

- Would you like to RUN this as a program? (it then marks the file +x)
- Shall I add this to your menu? (then so adds it...)
- It then moves the appImage to a sensible place (i.e. out of ~/Downloads)

It would make a VERY nice little addition to Mint.

**Comment by jbearden0** [comment]
This is my first post here, and it was to address this issue whatever works best to make using AppImages smoother. I agree with the other folks that fits in with the Linux Mint philosophy.
